### ISSUE TITLE
 UnicodeDecodeError: 'ascii' codec can't decode byte 0xe4 in position 14

### ISSUE THREAD

**Issue by pythonjian** [ansible/ansible#16269]
hi
  when i run ansible in Chinese folder , i got this error:
  An exception occurred during task execution. To see the full traceback, use -vvv. The error was: UnicodeDecodeError: 'ascii' codec can't decode byte 0xe4 in position 14: ordinal not in range(128)

  my path:
  /home/test/下载

 the error message:
  test@test:~/下载$ sudo ansible test -m copy -a "src=./config dest=/tmp/"
  An exception occurred during task execution. To see the full traceback, use -vvv. The error was: UnicodeDecodeError: 'ascii' codec can't decode byte 0xe4 in position 14: ordinal not in range(128)
192.168.1.83 | FAILED! => {
    "failed": true, 
    "msg": "Unexpected failure during module execution.", 
    "stdout": ""
}

  thanks

**Comment by abadger** [comment]
I was unable to reproduce this.  Can you post the version of ansible that you're using (ouput of ansible --version) and the output of your command with -vvvv?  The increased verbosity will let me see things like what locale settings are being used.

**Comment by alikins** [comment]
Need 'ansible --version' output and any other version or environment information.

Also need a copy of the output from ansible when run with '-vvvvv' verbose logging.
For example:

```
        ansible -vvvvv test -m copy -a "src=./config dest=/tmp/"
```

Other needed info is the current locale settings. The output of the 'locale' command for example.

```
   locale
```

That output should look something like:

```
LANG="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL=
```

Also, since you are running ansible via 'sudo', output from:

```
    sudo locale
```

[Note, it's generally not required to run 'ansible' itself as root or sudo...]

**Comment by pythonjian** [comment]
ansible version:

```
nerdguo@nerdguo:~/test/下载$ ansible --version
ansible 2.1.0.0
  config file = /home/nerdguo/.ansible.cfg
  configured module search path = Default w/o overrides
```

log:

```
nerdguo@nerdguo:~/test/下载$ sudo ansible -vvvvv test -m copy -a "src=./config dest=/tmp/"
Using /home/nerdguo/.ansible.cfg as config file
Loaded callback minimal of type stdout, v2.0
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 124, in run
    res = self._execute()
  File "/usr/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 446, in _execute
    result = self._handler.run(task_vars=variables)
  File "/usr/lib/python2.7/dist-packages/ansible/plugins/action/copy.py", line 104, in run
    source = self._loader.path_dwim_relative(self._loader.get_basedir(), 'files', source)
  File "/usr/lib/python2.7/dist-packages/ansible/parsing/dataloader.py", line 254, in path_dwim_relative
    search.append(self.path_dwim(os.path.join(basedir, dirname, source)))
  File "/usr/lib/python2.7/posixpath.py", line 80, in join
    path += '/' + b
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe4 in position 19: ordinal not in range(128)

10.2.1.83 | FAILED! => {
    "failed": true, 
    "msg": "Unexpected failure during module execution.", 
    "stdout": ""
}
```

locale:

```
nerdguo@nerdguo:~/test/下载$ sudo locale
LANG=zh_CN.UTF-8
LANGUAGE=zh_CN:zh
LC_CTYPE="zh_CN.UTF-8"
LC_NUMERIC="zh_CN.UTF-8"
LC_TIME="zh_CN.UTF-8"
LC_COLLATE="zh_CN.UTF-8"
LC_MONETARY="zh_CN.UTF-8"
LC_MESSAGES="zh_CN.UTF-8"
LC_PAPER="zh_CN.UTF-8"
LC_NAME="zh_CN.UTF-8"
LC_ADDRESS="zh_CN.UTF-8"
LC_TELEPHONE="zh_CN.UTF-8"
LC_MEASUREMENT="zh_CN.UTF-8"
LC_IDENTIFICATION="zh_CN.UTF-8"
LC_ALL=
```

**Comment by abadger** [comment]
This seems to be fixed in devel (which will be 2.2.0).  Figuring out whether we want to backport a fix or not.

**Comment by abadger** [comment]
Decided  that this is a rather invasive fix so we're not going to fix it in 2.1.x.  It will be in 2.2.0 when it comes out, though.
